pkg/estimator/server/framework/interface.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+
+	"k8s.io/client-go/informers"
+
+	"github.com/karmada-io/karmada/pkg/estimator/pb"
+)
+
+// Framework manages the set of plugins in use by the estimator.
+type Framework interface {
+	Handle
+	RunEstimateReplicasPlugins(ctx context.Context, replicaRequirements *pb.ReplicaRequirements) (int32, error)
```

Comment:
this inferface returns a `int32` value, I suggest add a simple comment to describe what does the numeric return value mean， though I known it represent the calculated replicas.

just for more readable, like another interface in `Karmada`:

https://github.com/chaosi-zju/karmada/blob/d017467fed08d24b1c5622604edccd23cd68e1aa/pkg/scheduler/framework/interface.go#L175-L182